### PR TITLE
chore: Add lint-staged to husky hook

### LINKED ISSUE #442: Lint only staged files on commit to reduce commit time

### PR DESCRIPTION
**Pull request by mayank99**
Added `lint-staged` v12 and upgraded `husky` to v4 (didn't do v7 because it's like a completely different library).
Autofixing on commit seems to be working but I can't tell if the commit time is faster. 🤷‍♀️

Closes #442

### CODE REVIEW

**Comment by mayank99** [comment]
I'm not sure why `yarn lint` is failing now. I didn't touch eslint at all. Will need to investigate.

**Comment by bentleyvk** [comment]
Huh?
image

File: package.json [hunk 1]
```diff
@@ -81,14 +81,15 @@
...
+    "husky": "^4.0.0",
     "inquirer": "^6.2.2",
     "jest": "^27.0.6",
     "jest-cli": "^27.0.6",
     "jest-junit": "^12.0.0",
+    "lint-staged": "^12.1.2",
     "markdown-to-jsx": "6.11.4",
     "mkdirp": "^1.0.4",
-    "prettier": "^2.2.1",
+    "prettier": "^2.5.0",
```
Comment (bentleyvk) [inline package.json:92]: Can you try reverting prettier version changes and checking whether build passes? Maybe they did some breaking changes or added some rules that we need to override.

Comment (mayank99) [reply package.json:92]: Bingo, that fixed it

Comment (bentleyvk) [reply package.json]: Can you check what has been changed if we need to update to the newest version for some reason?

Comment (mayank99) [reply package.json]: I'm not sure. I tried to check their release notes but didn't see anything https://prettier.io/blog/

**Review by bentleyvk — APPROVED** [review]

**Pull request merged**